src/htmx.js

```diff
@@ -127,14 +136,8 @@ return (function () {
         function getClosestAttributeValue(elt, attributeName) {
             var closestAttr = null;
             getClosestMatch(elt, function (e) {
-                closestAttr = getAttributeValue(e, attributeName)
-                if (closestAttr && elt !== e) {
-                  var inheritAttr = getAttributeValue(e, "hx-inherit");
-                  if (inheritAttr && (inheritAttr === "false" || inheritAttr.includes(attributeName))) {
-                      closestAttr = "unset";
-                  }
-                }
-                return closestAttr;
+                closestAttr = getAttributeValue(e, attributeName);
+                return closestAttr = getAttributeInheritance(elt, e, attributeName) || closestAttr;
```

Comment:
Other than the documentation states, this will allow for inheritance to be disabled by an element that does not necessarily specify the htmx-attribute it disables inheritance for.

```javascript
<div hx-select="#select">
    <div hx-target="#target" hx-inherit="hx-select">
        <!-- Won't have hx-select -->
        <button hx-get="/test/"></button>
    </div>
</div>
```

However this could be the desired behavior. Leaving it up to you. Otherwise we'll just have to wrap `getAttributeInheritance` in `if (closestAttr)`